devolutions-session/src/dvc/task.rs

```diff
@@ -830,6 +841,49 @@ async fn handle_exec_error(dvc_tx: &WinapiSignaledSender<NowMessage<'static>>, s
     }
 }
 
+fn set_kbd_layout(layout: SetKbdLayoutOption) -> anyhow::Result<()> {
+    // SAFETY: No outstanding preconditions.
+    let foreground_window = unsafe { GetForegroundWindow() };
+
+    if foreground_window.is_invalid() {
+        bail!("Failed to get foreground window handle");
+    }
+
+    let locale = match layout {
+        SetKbdLayoutOption::Next => isize::try_from(HKL_NEXT).expect("HKL_NEXT fits into isize"),
+        SetKbdLayoutOption::Prev => isize::try_from(HKL_PREV).expect("HKL_PREV fits into isize"),
+        SetKbdLayoutOption::Specific(layout) => {
+            // IMPORTANT: WM_INPUTLANGCHANGEREQUEST message only respects low word of the layout
+            // (locale) and high word (device) should be zero. Non-zero high word could produce
+            // unexpected results (e.g. switching to next layout instead of specific one).
+            //
+            // Loading locale via LoadKeyboardLayoutW and passing it to WM_INPUTLANGCHANGEREQUEST
+            // will not work as expected (even if dozen of sources suggest this approach) and
+            // easily breaks with some layouts.
```

Comment:
question/note: Does the `WPARAM(0)` below refers to the "high word (device)" of this comment? Maybe there is a better location for this commit.

Reply:
No, WPARAM contains flags (which are not relevant to us) and should be 0.